golink.go

```diff
@@ -169,17 +170,65 @@ func Run() error {
 	if err := srv.Start(); err != nil {
 		return err
 	}
+
 	localClient, _ = srv.LocalClient()
+out:
+	for {
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+		status, err := srv.Up(ctx)
+		if err == nil && status != nil {
+			break out
+		}
+	}
 
-	l80, err := srv.Listen("tcp", ":80")
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
nit: these type of contexts have a tendency of getting reused in the future at the wrong time. consider renaming to `statusCtx` for clarity